pkg/api/rest/delete.go

```diff
@@ -89,5 +89,6 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 	now := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
 	objectMeta.DeletionTimestamp = &now
 	objectMeta.DeletionGracePeriodSeconds = options.GracePeriodSeconds
+	objectMeta.Generation = objectMeta.Generation + 1
```

Comment:
Actually doing it here is not enough. We set DeletionTimestamp [here](https://github.com/kubernetes/kubernetes/blob/35922bdcbd087d25f7d342275d009c4493b68e21/pkg/registry/generic/registry/store.go#L569) as well.